---
layout: default
title: Reading Materials for RNA-seq Data Analysis
---

Below you can find the papers with additional background information on the tools that we discussed in the lectures. 

## RNA-seq Data Analysis: Basics

- RNA-seq review paper: [Van den Berge et al. (2018). RNA sequencing data: hitchhiker's guide to expression analysis. Peer J. Preprint](https://peerj.com/preprints/27283/)
- Sequencing does not remove biological variability: [Hansen et al. (2011). Sequencing technology does not eliminate biological variability. Nature biotechnology, 29(7): 572–573.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137276/)
- Core paper on edgeR: [McCarthy et al. (2012). Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation. Nucleic Acids Research, 40(10), 4288–4297](https://academic.oup.com/nar/article/40/10/4288/2411520)
- Core paper on DESeq2: [Love et al. (2016). Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 15:550](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8)
- Core paper on Limma-voom: [Law et al. (2014). voom: precision weights unlock linear model analysis tools for RNA-seq read counts. Genome Biology, 15:R29](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29)

## Normalisation
- TMM: [Robinson et al. (2010). A scaling normalization method for differential expression analysis of RNA-seq data. Genome Biology, 11:R25](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25)
- CQN: [Hansen et al. (2012). Removing technical variability in RNA-seq data using conditional quantile normalization. Biostatistics, 3(2):204-16](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3297825/)

## Stage-wise testing: 

- StageR paper: [Van den Berge et al. (2017). stageR: a general stage-wise method for controlling the gene-level false discovery rate in differential expression and differential transcript usage. Genome Biology, 18:151](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1277-0)

## Single-cell RNA-seq: 

- Zero-inflation: [Van den Berge et al. (2018). Observation weights unlock bulk RNA-seq tools for zero inflation and single-cell applications. Genome Biology, 19:24.](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1406-4)

## Transcript level alignment and inference: 

- Transcript level counting improves gene level inference: [Soneson et al. (2015). Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences.  F1000Research, 4:1521](https://f1000research.com/articles/4-1521/v2)
- Transcript level analysis: [Love et al. (2018). Swimming downstream: statistical analysis of differential transcript usage following Salmon quantification. F1000Research, 7:952](https://f1000research.com/articles/7-952/v3)

